src/System.Management.Automation/engine/regex.cs

```diff
@@ -345,13 +353,18 @@ public static string Unescape(string pattern)
 
             string s = null;
 
-            if (tempIndex > 0)
+            if (tempIndex == 0)
+            {
+                s = string.Empty;
+            }
+
+            if (tempIndex == pattern.Length)
             {
-                s = new string(temp, 0, tempIndex);
+                s = pattern;
```

Comment:
Do we have tests cases for regex escape?

Reply:
Only indirectly - the API is used in engine and involved in paths, completions and other tests.
I open new tracking issue #10051 to add tests for WildcardPattern class.